### PR TITLE
pkg/tinydtls: update version to fix compiler errors

### PR DESCRIPTION
**Pull request by smlng**
fixes compile error revealed in #8029

### CODE REVIEW

**Comment by smlng** [comment]
Murdock [output](https://ci.riot-os.org/RIOT-OS/RIOT/8029/8bc85b78695fc4cc068224d1d81965d9a32ccb62/output.html) for reference

File: pkg/tinydtls/Makefile.include [hunk 1]
```diff
@@ -2,6 +2,8 @@ PKG_BUILDDIR ?= $(PKGDIRBASE)/tinydtls
 
 INCLUDES += -I$(PKG_BUILDDIR)
 
+CFLAGS += -Wno-unused-parameter
```
Comment (kaspar030) [inline pkg/tinydtls/Makefile.include]: Does this have to be in Makefile.include? That way it gets applied globally. Maybe "Makefile" is a better choice?

**Comment by kaspar030** [comment]
Please squash.

**Comment by rfuentess** [comment]
For me, this PR is OK. I have been trying to put all the tinydtls unused variables with `(void)` statements, but it seems there are still some of them roaming in the code.

Probably it's worth to mention the discovered of @tobhe about [implicit fallthrough warnings](https://github.com/RIOT-OS/RIOT/pull/7615#issuecomment-330844825)

I have not had enough time for testing this issue, and because the warnings are coming from files that are under another type of license probably is not feasible to fix this warning in the source code (or is going to be slow).

File: pkg/tinydtls/Makefile.include [hunk 2]
```diff
@@ -3,7 +3,7 @@ PKG_BUILDDIR ?= $(PKGDIRBASE)/tinydtls
 INCLUDES += -I$(PKG_BUILDDIR)
 
 ifeq ($(TOOLCHAIN), llvm)
-  CFLAGS += -Wno-gnu-zero-variadic-macro-arguments -Wno-unused-function
+  CFLAGS += -Wno-gnu-zero-variadic-macro-arguments -Wno-unused-function -Wno-unused-parameter
```
Comment (kaspar030) [inline pkg/tinydtls/Makefile.include]: I'm sorry, if you touch this line, it'll also have to go into Makefile.

Comment (smlng) [reply pkg/tinydtls/Makefile.include]: somehow tinydtls only fails on native, and setting this in `Makefile` doesn't help? @kaspar030 any idea why, for libcoap it was okay to simply set it in `Makefile`.

Comment (smlng) [reply pkg/tinydtls/Makefile.include]: btw. it only fails if `DEVELHELP` is **not** set for native, if it is enabled the example builds.

Comment (smlng) [reply pkg/tinydtls/Makefile.include]: ah DEBUG output, I guess

Comment (smlng) [reply pkg/tinydtls/Makefile.include]: with DEVELHELP the parameter is used for some debug message and without its unused, arrrggg

**Comment by smlng** [comment]
I found another way to fix the compiler issues: simply update the package to latest master commit :)

**Comment by smlng** [comment]
required fix provided upstream see rfuentess/TinyDTLS#2

**Comment by rfuentess** [comment]
> required fix provided upstream see rfuentess/TinyDTLS#2

Done and ACK

**Comment by smlng** [comment]
@kaspar030 indirectly addressed your comments, but choosing a different path, i.e., bumping packet version and thereby fix issues - do you approve?

**Comment by kaspar030** [comment]
> do you approve?

Yes!

**Pull request merged**